Adding fintrospect example

daviddenton opened the pull request:
Framework homepage: is http://fintrospect.io 

Saw the post on reddit/r/scala re: web-frameworks, so thought I'd add an example. As well as the usual templating/JSON lib support, there are a few cool features that I don't believe are available elsewhere, such as re-using the HTTP contracts to generate type-safe HTTP clients, and auto-generating Swagger docs with matching Json Schema.

Fintrospect got 5th in JSON latest TechEmpower benchmarks (beating Http4s/Akka/Finch), so hopefully that's good enough to get listed here :)

https://www.techempower.com/benchmarks/#section=data-r13&hw=ph&test=json&l=4ftbsv

mfirry commented:
Awesome! I wonder how I missed this. Thanks.

mfirry commented:
On a second thought I don't think you've added both the required endpoints.

- a '/' endpoint that would return [1, 2, 3] (json)
- a '/:string' that would return "string" (for any string) (plaintext)

I only see the json one and actually it doesn't give out the expected result.

Would you mind fixing this?

Also, I do not need the client part.

Thanks so much!

daviddenton commented:
Sorry - I wasn't aware that you were checking the exact contract. 

Both endpoints were there, but I put the "echo" at /echo/<bob>, so it didn't clash with the generated swagger documentation. I've moved it now to the root and removed the swagger route (as it only gets used as a fallback if no other routes match). The JSON endpoint wasn't working due to the number types being converted to Number instead of a Int/Long - that's fixed now.

See PR #11 for the fixes

Thanks!
